idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+    "    organic Phase\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def build(self):\n",
+    "        \"\"\"\n",
+    "        Callable method for Block construction.\n",
+    "        \"\"\"\n",
+    "        super().build()\n",
+    "\n",
+    "        self._state_block_class = OrgPhaseStateBlock\n",
+    "\n",
+    "        # List of valid phases in property package\n",
+    "        self.Org = LiquidPhase()\n",
+    "\n",
+    "        # Component list - a list of component identifiers\n",
+    "        self.NaCl = Solute()\n",
+    "        self.KNO3 = Solute()\n",
+    "        self.CaSO4 = Solute()\n",
+    "        self.solvent = (\n",
+    "            Solvent()\n",
+    "        )  # Solvent used here is ethylene dibromide (Organic Polar)\n",
+    "\n",
+    "        self.solutes = Set(initialize=[\"NaCl\", \"KNO3\", \"CaSO4\"])\n",
+    "        # Heat capacity of solvent\n",
+    "        self.cp_mass = Param(\n",
+    "            mutable=True,\n",
+    "            initialize=717.01,\n",
+    "            doc=\"Specific heat capacity of solvent\",\n",
+    "            units=units.J / units.kg / units.K,\n",
+    "        )\n",
+    "\n",
+    "        self.dens_mass = Param(\n",
+    "            mutable=True,\n",
+    "            initialize=2170,\n",
+    "            doc=\"Density of ethylene dibromide\",\n",
+    "            units=units.kg / units.m**3,\n",
+    "        )\n",
+    "        self.temperature_ref = Param(\n",
+    "            within=PositiveReals,\n",
+    "            mutable=True,\n",
+    "            default=298.15,\n",
+    "            doc=\"Reference temperature\",\n",
+    "            units=units.K,\n",
+    "        )\n",
+    "        self.diffusion_factor = Param(\n",
+    "            self.solutes,\n",
+    "            initialize={\"NaCl\": 2.15, \"KNO3\": 3, \"CaSO4\": 1.5},\n",
+    "            within=PositiveReals,\n",
+    "            mutable=True,\n",
+    "        )\n",
+    "\n",
+    "    @classmethod\n",
+    "    def define_metadata(cls, obj):\n",
+    "        obj.add_default_units(\n",
+    "            {\n",
+    "                \"time\": units.hour,\n",
+    "                \"length\": units.m,\n",
+    "                \"mass\": units.g,\n",
+    "                \"amount\": units.mol,\n",
+    "                \"temperature\": units.K,\n",
+    "            }\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c942c2b",
+   "metadata": {},
+   "source": [
+    "# 1.3 State Block\n",
+    "\n",
+    "After the `Physical Parameter Block` class has been created, the next step is to write the code necessary to create the State Blocks that will be used throughout the flowsheet. `StateBlock` contains all the information necessary to define the state of the system. This includes the state variables and constraints on those variables which are used to describe a state property like the enthalpy, material balance, etc. \n",
```

Comment:
`PhysicalParameterBlock` shouldn't have spaces, check that it matches the object name. Similar for other object references in this notebook.